src/treemap/index.js

```diff
@@ -153,66 +136,79 @@ class Treemap extends React.Component {
       }
       return mappedNodes;
     }
-    if (data && mode === 'circlePack') {
+    if (mode === 'circlePack') {
       const packingFunction = pack()
-          .size([width, height])
+          .size([innerWidth, innerHeight])
           .padding(padding);
       const structuredInput = hierarchy(data)
         .sort((a, b) => a.size - b.size)
         .sum(d => d.size);
       return packingFunction(structuredInput).descendants();
     }
-    if (data) {
-      const tileFn = TREEMAP_TILE_MODES[mode];
-      const treemapingFunction = treemap(tileFn)
-        .tile(tileFn)
-        .size([width, height])
-        .padding(padding);
-      const structuredInput = hierarchy(data)
-        .sort((a, b) => a.size - b.size)
-        .sum(d => d.size);
 
-      return treemapingFunction(structuredInput).descendants();
-    }
-    return [];
+    const tileFn = TREEMAP_TILE_MODES[mode];
+    const treemapingFunction = treemap(tileFn)
+      .tile(tileFn)
+      .size([innerWidth, innerHeight])
+      .padding(padding);
+    const structuredInput = hierarchy(data)
+      .sort((a, b) => a.size - b.size)
+      .sum(d => d.size);
+
+    return treemapingFunction(structuredInput).descendants();
+
   }
 
   render() {
-    const {animation, className, height, mode, width} = this.props;
+    const {renderMode} = this.props;
+    const {scales} = this.state;
     const nodes = this._getNodesToRender();
-    const useCirclePacking = mode === 'circlePack';
-    return (
-      <div
-        className={`rv-treemap ${useCirclePacking ? 'rv-treemap-circle-packed' : ''} ${className}`}
-        style={{
-          width: `${width}px`,
-          height: `${height}px`
-        }}>
-        {nodes.map((node, index) => {
-          // throw out the rootest node
-          if (!useCirclePacking && !index) {
-            return null;
-          }
-
-          const nodeProps = {
-            animation,
-            node,
-            ...this.props,
-            x0: useCirclePacking ? node.x : node.x0,
-            x1: useCirclePacking ? node.x : node.x1,
-            y0: useCirclePacking ? node.y : node.y0,
-            y1: useCirclePacking ? node.y : node.y1,
-            r: useCirclePacking ? node.r : 1,
-            scales: this.state.scales
-          };
-          return (<TreemapLeaf {...nodeProps} key={`leaf-${index}`} />);
-        })}
-      </div>
-    );
+
+    if (renderMode === 'SVG') {
```

Comment:
maybe do something like

```js
const TreemapElement = renderMode === 'SVG' ? TreemapSVG : TreemapDOM;
return (
  <TreemapElement {...this.props} nodes={nodes} scales={scales} />
);
```

to prevent duplication?